api/src/main/java/org/jboss/forge/roaster/model/util/Types.java

```diff
@@ -298,7 +298,12 @@ public static boolean isGeneric(String type)
       StringTokenizer tok = new StringTokenizer(typeArgs, ", ");
       while (tok.hasMoreTokens())
       {
-         if (!validateNameWithGenerics(tok.nextToken()))
+         String typeArg = tok.nextToken();
+         while (incompleteGenerics(typeArg) && tok.hasMoreElements()) {
+            typeArg += tok.nextToken();
```

Comment:
Yes, that's why we should call toString after the iteraction, and probably refactor your new method to accept a StringBuilder instead. Thoughts?